### ISSUE TITLE
SQLSTATE[23000]: Integrity constraint violation: 1052 Column 'structureId' in where clause is ambiguous

### ISSUE THREAD

**Issue by ishetnogferre** [spicywebau/craft-neo#834]
### Bug Description

When trying to open a certain entry in the CMS (not all entries of the section have this, still trying to pinpoint what's different between them), we get the following error:
SQLSTATE[23000]: Integrity constraint violation: 1052 Column 'structureId' in where clause is ambiguous

Full error:
```
SQLSTATE[23000]: Integrity constraint violation: 1052 Column 'structureId' in where clause is ambiguous
The SQL being executed was: SELECT `elements`.`id`, `elements`.`canonicalId`, `elements`.`fieldLayoutId`, `elements`.`uid`, `elements`.`enabled`, `elements`.`archived`, `elements`.`dateLastMerged`, `elements`.`dateCreated`, `elements`.`dateUpdated`, `elements_sites`.`id` AS `siteSettingsId`, `elements_sites`.`slug`, `elements_sites`.`siteId`, `elements_sites`.`uri`, `elements_sites`.`enabled` AS `enabledForSite`, `elements`.`dateDeleted`, `neoblocks`.`fieldId`, `neoblocks`.`primaryOwnerId`, `neoblocks`.`typeId`, `neoblocks_owners`.`ownerId`, `neoblocks_owners`.`sortOrder`, `content`.`id` AS `contentId`, `content`.`field_commonAddress_lgbisgkj`, `content`.`field_projectArchitect_ititmtic`, `content`.`field_projectArea_qqrlatlc`, `content`.`field_aspectRatio_pecyaiwn`, `content`.`field_commonBackgroundColor_sbtaaaxv`, `content`.`field_cardColor_nltjrdsq`, `content`.`field_isCentered_fugodqth`, `content`.`field_agentCity_koqurovg`, `content`.`field_projectCity_kwgjcugl`, `content`.`field_commonColorPicker_xdlhqdqq`, `content`.`field_column_uzizfebh`, `content`.`field_agentCompany_rmetuavz`, `content`.`field_projectCompany_bakezdxi`, `content`.`field_isContainer_vhxvgzvc`, `content`.`field_commonCountryCode_fcpzfick`, `content`.`field_commonDate_drfdhrtr`, `content`.`field_commonDownloadables_sbmuawhh`, `content`.`field_agentEmail_yykghsoq`, `content`.`field_commonMail_bfuuhxmy`, `content`.`field_agentFirstName_blbsuqfn`, `content`.`field_commonIcon_zflzlofg`, `content`.`field_imageFit_ioduomlr`, `content`.`field_itemSize_ykddtclp`, `content`.`field_agentLanguage_lodcruoi`, `content`.`field_agentLastName_ocmiqmhu`, `content`.`field_commonPrimaryLink_xktrajtb`, `content`.`field_commonSecondaryLink_tanwmpvt`, `content`.`field_productMaintenance_farqglax`, `content`.`field_productMinQty_vkwysnse`, `content`.`field_isNew_guzyuzjp`, `content`.`field_oldId_gwwlplmh`, `content`.`field_optimisedFullWidthImages_ymjaksno`, `content`.`field_optimisedLogoImages_oncvwnkj`, `content`.`field_optimisedPortraitImages_opuacywo`, `content`.`field_optimisedRectangleUpImages_xngcydfn`, `content`.`field_optimisedSplitImages_wmxfqwwq`, `content`.`field_optimisedSquareImages_qnljlksi`, `content`.`field_optimisedWidescreenImages_djrpciog`, `content`.`field_overrideDefaults_srrvzpjg`, `content`.`field_agentPhoneNumber_riutncwz`, `content`.`field_photoFilter_jizodegn`, `content`.`field_projectPhotographer_djnybqij`, `content`.`field_isPopular_vrobqgor`, `content`.`field_commonPosition_scqksycz`, `content`.`field_productPricing_dkiwuxsp`, `content`.`field_productGroupPicker_gzxfmxea`, `content`.`field_salesNotifText_kozeagzk`, `content`.`field_commonScript_vovhdkbx`, `content`.`field_isSeamless_uxfmfrux`, `content`.`field_commonSEO_bfzjaleo`, `content`.`field_showInFilters_fzdislku`, `content`.`field_showInMenu_elrxaoau`, `content`.`field_commonSize`, `content`.`field_commonSubtitle_qunyxqwi`, `content`.`field_teaserDate_dxcuexsf`, `content`.`field_teaserSubtitle_xihfxssa`, `content`.`field_teaserText_tisonoui`, `content`.`field_teaserTitle_jwdwkvlm`, `content`.`field_productTechInfo_vnvlaamo`, `content`.`field_commonTelephone_hkhdpfqs`, `content`.`field_commonText_msrihima`, `content`.`field_hasDarkBg_pkcevjzn`, `content`.`field_commonTextRich_ncpiazdl`, `content`.`field_commonTitle_lbayszem`, `structureelements`.`root`, `structureelements`.`lft`, `structureelements`.`rgt`, `structureelements`.`level`, `structureelements`.`structureId`
FROM (SELECT `elements`.`id` AS `elementsId`, `elements_sites`.`id` AS `elementsSitesId`, `content`.`id` AS `contentId`, `structureelements`.`structureId`
FROM `elements` `elements`
INNER JOIN `neoblocks` `neoblocks` ON `neoblocks`.`id` = `elements`.`id`
INNER JOIN `neoblocks_owners` `neoblocks_owners` ON (`neoblocks_owners`.`blockId` = `elements`.`id`) AND (`neoblocks_owners`.`ownerId`=57174)
INNER JOIN `elements_sites` `elements_sites` ON `elements_sites`.`elementId` = `elements`.`id`
INNER JOIN `content` `content` ON (`content`.`elementId` = `elements`.`id`) AND (`content`.`siteId` = `elements_sites`.`siteId`)
LEFT JOIN `structureelements` `structureelements` ON (`structureelements`.`elementId` = `elements`.`id`) AND (EXISTS (SELECT *
FROM `structures` use index(primary)
WHERE (`id` = `structureelements`.`structureId`) AND (`dateDeleted` IS NULL)))
INNER JOIN `neoblockstructures` `neoblockstructures` ON `neoblockstructures`.`structureId` = `structureelements`.`structureId`
WHERE (`neoblocks`.`fieldId`=113) AND (NOT (`structureId` IS NULL)) AND (`elements_sites`.`siteId`=1) AND (`elements`.`archived`=FALSE) AND (`elements`.`draftId` IS NULL) AND (`elements`.`revisionId` IS NULL) AND (`neoblockstructures`.`fieldId`=113) AND (`neoblockstructures`.`ownerId`=57174) AND (`neoblockstructures`.`siteId`=1)
ORDER BY `structureelements`.`lft`, `neoblocks_owners`.`sortOrder`) `subquery`
INNER JOIN `elements` `elements` ON `elements`.`id` = `subquery`.`elementsId`
INNER JOIN `elements_sites` `elements_sites` ON `elements_sites`.`id` = `subquery`.`elementsSitesId`
INNER JOIN `neoblocks` `neoblocks` ON `neoblocks`.`id` = `subquery`.`elementsId`
INNER JOIN `neoblocks_owners` `neoblocks_owners` ON (`neoblocks_owners`.`blockId` = `elements`.`id`) AND (`neoblocks_owners`.`ownerId`=57174)
INNER JOIN `content` `content` ON `content`.`id` = `subquery`.`contentId`
LEFT JOIN `structureelements` `structureelements` ON (`structureelements`.`elementId` = `subquery`.`elementsId`) AND (`structureelements`.`structureId` = `subquery`.`structureId`)
ORDER BY `structureelements`.`lft`, `neoblocks_owners`.`sortOrder`
```

Some extra lines that might help debug this:
<img width="1442" alt="Screenshot 2024-01-24 at 10 26 14" src="https://github.com/spicywebau/craft-neo/assets/6649706/acb3706f-78f0-4a30-a787-b5dd8fc50819">

### Steps to reproduce

(will update this if I have determined the difference between the entries of the same section having this)

### Neo version

4.0.2

### Craft CMS version

4.7.0

### What is the affected Neo field's propagation method?

Save blocks to all sites the owner element is saved in

### Does this issue involve templating, and if so, is eager-loading used?

This is not a templating issue

**Comment by Gawel91** [comment]
Same issue here, it happened also in Neo 4.0.0 and after update to 4.0.2
The issue happens to only one entry.

**Comment by ttempleton** [comment]
@ishetnogferre @Gawel91 If possible, could one of you please send your composer files and database backup to plugins@spicyweb.com.au, along with instructions for which entry this happens with, and we'll have a look at it.

**Comment by acantes** [comment]
Same issue here. I have just sent my file to you @ttempleton as requested.

**Comment by ttempleton** [comment]
Thanks for sending that @acantes - the error should be fixed now in Neo 4.0.3.
